### ISSUE TITLE
Is it possible to remove some DOM elements from Persona and PersonaCoin?

### ISSUE THREAD

**Issue by spmonahan** [microsoft/fluentui#23386]
## Problem

In scenarios where applications have many Personas rendered the size of the DOM can become quite large. For example, rendering 235 Personas generates approximately 3,500 DOM nodes.

For example, this rendered element:

![A rendered persona](https://user-images.githubusercontent.com/93940821/171757056-65f8926b-22e9-4c9e-9735-d9ce481c1738.png)

Generates this DOM:

![DOM rendered for persona](https://user-images.githubusercontent.com/93940821/171757216-1f235def-c102-4e69-91ad-dd9e89b503ef.png)

## Solution

We'd like to remove DOM nodes and only render things that are absolutely required. This can either be achieved by removing DOM nodes entirely or by conditionally rendering nodes.

On the topic of conditionally rendering nodes, the hidden Tooltip nodes from the screenshot seem like good candidates but they need to be rendered to work properly with iOS VoiceOver. See #22216 for details.

## Risks (Rabbit holes)

Changing rendered DOM runs the risk of being a breaking change.

**Comment by behowell** [comment]
I've identified a few places that can be improved:
* The tooltips can be removed by setting `showOverflowTooltip={false}` on the Persona controls. These tooltips are there in case the text gets truncated, but don't appear to be necessary if a contact card is shown on hover.
* I've published a PR to stop rendering some unnecessary DOM nodes in Persona: #23401 
  * Removes extra empty `<div>` nodes rendered for tertiary and optional text, even if there is no tertiary/optional text
  * Removes an extra `<span>` surrounding the initials in PersonaCoin

With those changes implemented, it will remove 7 DOM nodes from the example in the issue description.

The DOM of Persona/PersonaCoin does still seem to have more divs than should be necessary, but it is difficult to change further without potentially breaking some users.

**Comment by KeiraYang** [comment]
@behowell `showOverflowTooltip` are not in the Fabric version 7 we're using. Can you cherry-pick the props change for it?

**Comment by behowell** [comment]
Hi @KeiraYang, I logged #23480 to track the request to backport the prop to v7.

**Comment by spmonahan** [comment]
To answer the question posed by the title, after Ben's changes, it is no longer possible to remove more DOM elements from Persona/PersonaCoin without introducing breaking changes. Because my recent work is focused on performance I've continued to look at this issue because I think there are three interesting questions that flow from this issue:

1. How much did Ben's change improve performance?
2. Assuming we could introduce breaking changes, what, if any, further performance gains might we see?
3. How much does v9 Persona improve performance relative to v8

More on that below, but generally my recommendation is that users move to Fluent UI v9 to realize improved performance. Fluent v9 represents the latest Fluent design language, is generally more performant than v8 and is the focus of most Fluent React development going forward.

I've run [two performance tests](https://github.com/microsoft/fluentui/tree/master/apps/stress-test) in Edge on Windows 11. Both tests load large DOM trees with complex CSS styles. One test, "inject styles" focuses on style recalculation, layout and paint performance. The other, "mount" includes style performance but also measures how long it takes for the first load of a component into the DOM. I've used the "XL" test fixture which is a tree with 4,000 DOM nodes (total nodes in the DOM will be 4,000 \* _number of nodes rendered by component_). Since we're removing DOM nodes and measuring perf changes the overall number of nodes in the DOM varies but the tests are in the range of 20,000+ DOM nodes.

So, how much did [Ben's changes](https://github.com/microsoft/fluentui/pull/23401/files#diff-7091c981a5e0e0b8243163dd9993e24a59d459921aa63ee9513e51e535f2f4e1) improve Persona performance? Style recalculation improved by 18.37%. Mount improved by 12.68%.

Taking Ben's changes a step further I removed an additional DOM node from PersonaCoin. The `div` removed is essentially a wrapper though it does have props, a classname and `styles` applied. Making this change does affect behavior in some cases (though not in the scenario under test) but leaves the same visual result as before the change. This changes improves style recalculation by 7.72% and mount by 6.01%. While this is still a nice improvement, given that it requires a breaking API change (and may not even be practical given that it removes some functionality) it doesn't really move the needle imo.

Looking at the recommendation to move to v9 we can compare v8 Persona with v9 Persona (which is in unstable preview as of this writing). In my test I compare a basic Persona usage which includes an Avatar/PersonaCoin image, a name, a secondary text (i.e., "available") and a presence statues (i.e., available/online). Comparing the XL fixtures v9 is a 20.7% improvement in style recalculation over v9 and a 19.9% improvement in mount performance.
